src/sema/functions.rs

```diff
@@ -355,19 +355,19 @@ pub fn contract_function(
         } else if func.body.is_none() {
             ns.diagnostics.push(Diagnostic::error(
                 func.loc,
-                "function in a library must have a body".to_string(),
+                format!("{} in a library must have a body", func.ty),
             ));
             success = false;
         } else if let Some((loc, _)) = is_override {
             ns.diagnostics.push(Diagnostic::error(
                 loc,
-                "function in a library cannot override".to_string(),
+                format!("{} in a library cannot override", func.ty),
             ));
             success = false;
         } else if let Some(pt::Mutability::Payable(_)) = mutability {
             ns.diagnostics.push(Diagnostic::error(
                 func.loc,
-                "function in a library cannot be payable".to_string(),
+                format!("{} in a library cannot be payable", func.ty),
             ));
```

Comment:
A sema test would make more sense in this case, so we can not only verify if we are allowing such a case, but also check that the error message is properly displayed.

Reply:
We have contract_tests for this, which ensure that the diagnostics and its location are correct. I've added some more tests so they're all covered.